scripts/pdsp/process.py

```diff
@@ -0,0 +1,306 @@
+
+        # Some part of the job output might be incomplete
+        # and we can exclude them.
+        # Can be a single string or a list of strings.
+        # If nothing to be exclude, the field can be omitted.
+        'exclude': '61503165_6',
+
+        # Template of input filenames in the job output folder.
+        # Each fields is contained in a bracket.
+        # The fields specified in the input_template must match
+        # the keys in the fields item below.
+        'input_template': 'frames[Sim]-[SigProc][APA]_*.npz',
+
+        # The values for each field specified in the input template.
+        'fields': {
+            'Sim': {'1d': 'fake', '2d': 'real'},
+            'SigProc': ('orig', 'gauss'),
+            'APA': (0, 1, 2, 3, 4, 5)
+        },
+
+        # Location to save the output.
+        'output_folder': '/data/datasets/LS4GAN_yuhw/full_images',
+
+        # The path template of each output file.
+        # The "Event" and "Plane" (U, V, W) are parsed from each input field.
+        'output_template': '[SigProc]/[Sim]/JobID_[JobID]-Event_[Event]-[APA]-[Plane].npz'
+    }
+
+    To-does:
+    """
+    def __init__(self, config):
+
+        self.config = config
+
+        self.input_template = config['input_template']
+        self.fields = config['fields']
+
+        self.__verify()
+
+        # job id prefix(es) to load
+        if isinstance(config['JobID'], str):
+            self.job_ids = (config['JobID'],)
+        else:
+            self.job_ids = tuple(config['JobID'])
+
+        # process folders to exclude
+        self.excludes = ()
+        if 'exclude' in config:
+            if isinstance(config['exclude'], str):
+                self.excludes = (config['exclude'],)
+            else:
+                self.excludes = tuple(config['exclude'])
+
+        # set up output
+        self.output_folder = Path(config['output_folder'])
+        if self.output_folder.exists():
+            print(f"{config['output_folder']} is not empty.")
+            sys.exit("Refuse to overwrite an exist folder")
+        self.output_folder.mkdir(parents=True)
+        self.output_template = config['output_template']
+
+
+    def __verify(self):
+        """
+        Verify that values for all fields in the input template
+        has been provided in the config.
+        """
+        input_keys = re.findall(r"\[(\w+)\]", self.config['input_template'])
+        assert set(input_keys) == self.config['fields'].keys()
+
+
+    def __load_folders(self):
+        """
+        Load all job output folders prefixed by given job id(s)
+        from the input folder
+        """
+        # make sure the input folder exists
+        input_folder = Path(self.config['input_folder'])
+        assert input_folder.exists(), \
+            f"input folder {input_folder} doesn't exist"
+
+        # Locate all the job output folders
+        # that are not in exclusion list.
+        folders = {}
+        for job_id_glob in self.job_ids:
```

Comment:
If I follow the code correctly, `job_id_glob` is not actually a glob, but just a subdir name prefix?